src/main/java/org/jboss/xavier/analytics/functions/HelperFunctions.java

```diff
@@ -145,4 +151,34 @@ public static boolean isFlagUnsuitableForOCP(String flagToCheck)
     {
         return Arrays.stream(FlagUnsuitabilityForTargets.values()).anyMatch(value -> flagToCheck.toLowerCase().contains(value.getName().toLowerCase()) && value.isUnsuitableForOCP());
     }
+
+    public static boolean doesFlagsCollectionContainCriticalCategory(Set<String> flags) {
+        if (flags.stream().anyMatch(flag -> Arrays.stream(FlagUnsuitabilityForTargets.values()).anyMatch
+                (value -> flag.toLowerCase().contains(value.getName().toLowerCase()) && WorkloadInventoryReportModel.FLAG_CATEGORY_CRITICAL.equals(value.getCategoryLevel()))))
+        {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean doesFlagsCollectionContainWarningCategory(Set<String> flags) {
+        if (flags.stream().anyMatch(flag -> Arrays.stream(FlagUnsuitabilityForTargets.values()).anyMatch
+                (value -> flag.toLowerCase().contains(value.getName().toLowerCase()) && WorkloadInventoryReportModel.FLAG_CATEGORY_WARNING.equals(value.getCategoryLevel()))))
+        {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean doesFlagsCollectionContainInformationCategory(Set<String> flags) {
+        if (flags.stream().anyMatch(flag -> Arrays.stream(FlagUnsuitabilityForTargets.values()).anyMatch
+                (value -> flag.toLowerCase().contains(value.getName().toLowerCase()) && WorkloadInventoryReportModel.FLAG_CATEGORY_INFORMATION.equals(value.getCategoryLevel()))))
+        {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Since all 3 methods are almost the same, could we create just one single method and pass `WorkloadInventoryReportModel.FLAG_CATEGORY_INFORMATION, WorkloadInventoryReportModel.FLAG_CATEGORY_WARNING, or WorkloadInventoryReportModel.FLAG_CATEGORY_CRITICAL` as a parameter? wdyt?

Reply:
done